kratos/expression/entity_domain_size_expression_io.h

```diff
@@ -0,0 +1,89 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#pragma once
+
+// System incldues
+
+// Project includes
+#include "expression/container_expression.h"
+#include "expression/traits.h"
+#include "expression_io.h"
+#include "includes/define.h"
+#include "includes/model_part.h"
+
+namespace Kratos {
+
+
+class KRATOS_API(KRATOS_CORE) EntityDomainSizeExpressionIO
+{
+public:
+    ///@name Type definitions
+    ///@{
+
+    using Configuration = Kratos::Globals::Configuration;
+
+    ///@}
+    ///@name Public classes
+    ///@{
+
+    class KRATOS_API(KRATOS_CORE) EntityDomainSizeExpressionInput : public ExpressionInput
+    {
+    public:
+        ///@name Type definitions
+        ///@{
+
+        KRATOS_CLASS_POINTER_DEFINITION(EntityDomainSizeExpressionInput);
+
+        ///@}
+        ///@name Life cycle
+        ///@{
+
+        EntityDomainSizeExpressionInput(
+            const ModelPart& rModelPart,
+            const ContainerType& rContainerType);
+
+        ~EntityDomainSizeExpressionInput() override = default;
+
+        ///@}
+        ///@name Public operations
+        ///@{
+
+        Expression::Pointer Execute() const override;
+
+        ///@}
+
+    private:
+        ///@name Private member variables
+        ///@{
+
+        const ModelPart& mrModelPart;
+
+        const ContainerType mContainerType;
```

Comment:
I'd rather stick to [`Globals::DataLocation`](https://github.com/KratosMultiphysics/Kratos/blob/fbc41845e9059b8457c9fa44dd0d2457a1c48931/kratos/includes/global_variables.h#L47-L55) since it's already in core, and get rid of `ContainerType`.

Also, the `const` qualifier prevents assignment operators again.
```suggestion
        Globals::DataLocation mContainerType;
```

Reply:
done :)